### ISSUE TITLE
Display of 0-D datetime64 arrays is odd.

### ISSUE THREAD

**Issue by charris** [numpy/numpy#4337]
```
In [51]: b = array(datetime(2009, 1, 1, 0, 0), 'datetime64[us]')

In [52]: b
Out[52]: array(datetime.datetime(2009, 1, 1, 0, 0), dtype='datetime64[us]')

In [53]: array((b,b))
Out[53]: array(['2008-12-31T17:00:00.000000-0700', '2008-12-31T17:00:00.000000-0700'], dtype='datetime64[us]')

In [54]: b[()]
Out[54]: numpy.datetime64('2008-12-31T17:00:00.000000-0700')
```

**Comment by shoyer** [comment]
@charris What would you consider an appropriate repr for 0-D datetime64? Something more like `array('2000-01-01T00:00:00.000000', dtype='datetime64[us]')`?

**Comment by shoyer** [comment]
Even worse is repr for 0D datetime64 arrays with ns precision, which just shows integers:
```
In [53]: np.array(np.datetime64('2000-01-01', 'ns'))
Out[53]: array(946684800000000000, dtype='datetime64[ns]')
```

This seems possibly related to https://github.com/numpy/numpy/issues/7619 -- the repr on 0-dimensional datetime64 arrays seems to use the same code path as `.item()`.

**Comment by eric-wieser** [comment]
This is fixed in 1.14-dev by #8983 :

```python
>>> from datetime import datetime
>>> np.array(datetime(2009, 1, 1, 0, 0), 'datetime64[us]')
array('2009-01-01T00:00:00.000000', dtype='datetime64[us]')
>>> np.array(np.datetime64('2000-01-01', 'ns'))
array('2000-01-01T00:00:00.000000000', dtype='datetime64[ns]')
```
